### PR TITLE
Support timestamp in table query request in oss delta sharing server

### PR DESCRIPTION
**Pull request by linzhou-db**
Support timestamp in table query request in oss delta sharing server, only one of timestamp and version is supported, and they are only supported when cdfEnabled is true.

### CODE REVIEW

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala [hunk 1]
```diff
@@ -268,12 +269,13 @@ class DeltaSharingService(serverConfig: ServerConfig) {
 
     val start = System.currentTimeMillis
     val tableConfig = sharedTableManager.getTable(share, schema, table)
-    if (queryTableRequest.version.isDefined) {
+    if (queryTableRequest.version.isDefined || queryTableRequest.timestamp.isDefined) {
       if (!tableConfig.cdfEnabled) {
-        throw new DeltaSharingIllegalArgumentException("Reading table by version is not supported" +
-          s" because change data feed is not enabled on table: $share.$schema.$table")
+        throw new DeltaSharingIllegalArgumentException("Reading table by version or timestamp is" +
+          " not supported because change data feed is not enabled on table: " +
```
Comment (chakankardb) [inline server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:275]: Question: With streaming, we will allow time travel queries even if cdf is not enabled... right?

File: server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala [hunk 2]
```diff
@@ -133,10 +133,19 @@ class DeltaSharedTable(
...
       predicateHints: Seq[String],
       limitHint: Option[Long],
-      version: Option[Long]): (Long, Seq[model.SingleAction]) = withClassLoader {
+      version: Option[Long],
+      timestamp: Option[String]): (Long, Seq[model.SingleAction]) = withClassLoader {
     // TODO Support `limitHint`
-    val snapshot = if (version.isEmpty) deltaLog.snapshot else {
+    if (version.isDefined && timestamp.isDefined) {
+      throw new IllegalArgumentException("Please either provide '<version>' or '<timestamp>'")
+    }
```
Comment (chakankardb) [inline server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala:141]: Should we add a unit test for this?

**Review by chakankardb — APPROVED** [review]

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala (hunk 1 above)
Comment (linzhou-db) [reply server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:275]: Yes. We'll have another field to guard this, pending on the decision about default behavior change.

File: server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala (hunk 2 above)
Comment (linzhou-db) [reply server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala:141]: hmmm, missed again..

Comment (linzhou-db) [reply server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala:141]: I'll add it in #187

**Pull request merged**